yt/data_objects/data_containers.py

```diff
@@ -2179,3 +2182,93 @@ def _reconstruct_object(*args, **kwargs):
     obj = cls(*new_args)
     obj.field_parameters.update(field_parameters)
     return ReconstructedObject((ds, obj))
+
+def monte_carlo_sample(ds, n_samples=100000, fields=None, n_neighbors=1):
```

Comment:
I think if this is a top-level function, it should probably go into a different file, perhaps into a file all by itself, but it seems to me that making it a function on a data object might be a bit easier and could keep it in this file.

It would be nice to have it possible to specify the `field_type` to sample, too.

Reply:
Yeah, I'm dithering on whether or not it should be a `Dataset` method.  Honestly, I don't think it has enough widespread use to necessitate being attached to the `Dataset` object.  Plus, I'm not entirely sure how to only have it be on Grid-based datasets short of going through and adding it to each grid-based frontend.  So it may be best to just have it as a stand-alone function sitting somewhere, but not sure where.

Reply:
Oh, I meant data object -- so that you could just resample, say, a sphere.

Reply:
There is the little-used and not-documented particle generators stuff:

https://github.com/yt-project/yt/blob/master/yt/utilities/particle_generator.py

Reply:
Basically what you are doing is a better version of this class:

https://github.com/yt-project/yt/blob/master/yt/utilities/particle_generator.py#L328